src/Extractor/ContentExtractor.php

```diff
@@ -169,30 +169,9 @@ public function process($html, $url, SiteConfig $siteConfig = null, $smartTidy =
     {
         $this->reset();
 
-        $this->siteConfig = $siteConfig;
-        if (null === $this->siteConfig) {
-            $this->siteConfig = $this->buildSiteConfig($url, $html);
-        }
+        $this->prepareSiteConfig($html, $url, $siteConfig);
```

Comment:
Regarding scrutinizer issue, maybe `prepareSiteConfig` should return the siteconfig, so you can assign it to `$this->siteConfig`

https://scrutinizer-ci.com/g/j0k3r/graby/inspections/2f56b742-0594-4b71-849c-651337f94134/issues/files/src/Extractor/ContentExtractor.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

Reply:
Ah, I don't seem to be able to see the Scrutinizer report.

However, looking at the code, there was an incorrect `@return` in the docblock for this method. Rather than refactoring it to return something, I just removed that annotation. Let's see how it goes.